internal/services/springcloud/spring_cloud_service_resource.go

```diff
@@ -351,11 +351,18 @@ func resourceSpringCloudServiceCreate(d *pluginsdk.ResourceData, meta interface{
 	}
 	d.SetId(id.ID())
 
-	log.Printf("[DEBUG] Updating Config Server Settings for %s..", id)
-	if err := updateConfigServerSettings(ctx, configServersClient, id, gitProperty); err != nil {
-		return err
+	skuName := d.Get("sku_name").(string)
+	if skuName == "E0" && gitProperty != nil {
+		return fmt.Errorf("`config_server_git_setting` is not supported for sku `E0`")
+	}
+
+	if skuName != "E0" {
```

Comment:
and this should be an else?
```suggestion
	} else if gitProperty != nil {
```